src/Controller/Component/AuthComponent.php

```diff
@@ -402,14 +402,21 @@ protected function _unauthenticated(Controller $controller)
      */
     protected function _loginActionRedirectUrl()
     {
-        $currentUrl = $this->request->here(false);
+        $urlToRedirectBackTo = $this->request->here(false);
+        if (!$this->request->is('get')) {
+            $urlToRedirectBackTo = $this->request->referer(true);
```

Comment:
If we kept this in, it would at least cover the 90% case, whereas right now everyone would run into trouble.